circuit_maintenance_parser/parsers/aws.py

```diff
@@ -60,29 +76,63 @@ def parse_text(self, text):
             This maintenance is scheduled to avoid disrupting redundant connections at =
             the same time.
         """
-        data = {"circuits": []}
+        text_map = {
+            "^Account ?I?D?: ([0-9]+)": "account",
+            "^Start Time: ([A-Z][a-z]{2}, [0-9]{1,2} [A-Z][a-z]{2,9} [0-9]{4} [0-9]{2}:[0-9]{2}:[0-9]{2} [A-Z]{2,3})": "start",
+            "^End Time: ([A-Z][a-z]{2}, [0-9]{1,2} [A-Z][a-z]{2,9} [0-9]{4} [0-9]{2}:[0-9]{2}:[0-9]{2} [A-Z]{2,3})": "end",
+            "(?<=from )([A-Z][a-z]{2}, [0-9]{1,2} [A-Z][a-z]{2,9} [0-9]{4} [0-9]{2}:[0-9]{2}:[0-9]{2} [A-Z]{2,3}) to ([A-Z][a-z]{2}, [0-9]{1,2} [A-Z][a-z]{2,9} [0-9]{4} [0-9]{2}:[0-9]{2}:[0-9]{2} [A-Z]{2,3})": "start_and_end",
+        }
+
+        regex_keys = re.compile("|".join(text_map), re.IGNORECASE)
+
+        data = {"circuits": [], "start": "", "end": ""}
         impact = Impact.OUTAGE
         maintenace_id = ""
         status = Status.CONFIRMED
+
         for line in text.splitlines():
             if "planned maintenance" in line.lower():
                 data["summary"] = line
-                search = re.search(
-                    r"([A-Z][a-z]{2}, [0-9]{1,2} [A-Z][a-z]{2,9} [0-9]{4} [0-9]{2}:[0-9]{2}:[0-9]{2} [A-Z]{2,3}) to ([A-Z][a-z]{2}, [0-9]{1,2} [A-Z][a-z]{2,9} [0-9]{4} [0-9]{2}:[0-9]{2}:[0-9]{2} [A-Z]{2,3})",
-                    line,
-                )
-                if search:
-                    data["start"] = self.dt2ts(parser.parse(search.group(1)))
-                    data["end"] = self.dt2ts(parser.parse(search.group(2)))
-                    maintenace_id += str(data["start"])
-                    maintenace_id += str(data["end"])
+            # match against the regex strings
+            line_matched = re.search(regex_keys, line)
+            # if we have a string that's not in our text_map
+            # there may still be some strings with data to capture.
+            # otherwise, continue on.
+            if not line_matched:
                 if "may become unavailable" in line.lower():
                     impact = Impact.OUTAGE
                 elif "has been cancelled" in line.lower():
                     status = Status.CANCELLED
-            elif re.match(r"[a-z]{5}-[a-z0-9]{8}", line):
-                maintenace_id += line
-                data["circuits"].append(CircuitImpact(circuit_id=line, impact=impact))
+
+                if re.match(r"[a-z]{5}-[a-z0-9]{8}", line):
+                    maintenace_id += line
+                    data["circuits"].append(CircuitImpact(circuit_id=line, impact=impact))
+                continue
+
+            # for lines that do match our regex strings.
+            # grab the data and map the values to keys.
+            for group_match in line_matched.groups():
+                if group_match is not None:
```

Comment:
similar to above.

Reply:
nice! yes this let me shore up one indentation. I'm definitely needing to explore guardian clauses.